core/src/HIP/Kokkos_HIP_Parallel_Team.hpp

```diff
@@ -995,12 +1025,16 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
                   m_policy.thread_scratch_size(0)) /
                   m_vector_size;
 
+    // We can't early exit here because the result place might not be accessible
+    // or the functor/reducer init not callable on the host. But I am not sure
+    // all the other code below is kosher with zero work length ...
+    //
     // Return Init value if the number of worksets is zero
-    if (arg_policy.league_size() == 0) {
-      value_init::init(reducer_conditional::select(m_functor, m_reducer),
-                       m_result_ptr);
-      return;
-    }
+    // if (arg_policy.league_size() == 0) {
+    //   value_init::init(reducer_conditional::select(m_functor, m_reducer),
+    //                   m_result_ptr);
+    //  return;
+    //}
```

Comment:
Why are you less certain here than for CUDA? I am not aware that we do anything much differently here. Anyway, this comment should go away one way or the other.

Reply:
As a reminder to you to check whether there is something which could go wrong ;-)

Reply:
removed